src/django/api/facility_history.py

```diff
@@ -162,6 +162,60 @@ def processing_results_has_split_action_for_oar_id(list_item, facility_id):
     ]
 
 
+def create_facility_claim_entry(claim):
+    if claim.status == FacilityClaim.REVOKED:
+        return {
+            'updated_at': str(claim.updated_at),
+            'action': FacilityHistoryActions.CLAIM_REVOKE,
+            'detail': 'Claim on facility {} by {} was revoked'.format(
+                claim.facility.id,
+                claim.contributor.name,
+            ),
+        }
+
+    if claim.status == FacilityClaim.APPROVED \
+       and claim.prev_record.status == FacilityClaim.PENDING:
+        return {
+            'updated_at': str(claim.updated_at),
+            'action': FacilityHistoryActions.CLAIMED,
+            'detail': 'Facility {} was claimed by {}'.format(
+                claim.facility.id,
+                claim.contributor.name,
+            ),
+        }
+
+    if claim.status == FacilityClaim.APPROVED:
+        public_claim_data_keys = [
+            'facility_description',
+            'facility_name_english',
+            'facility_name_native_language',
+            'facility_minimum_order',
+            'facility_average_lead_time',
+            'facility_workers_count',
+            'facility_female_workers_percentage',
+            'facility_type',
+            'other_facility_type',
+            'facility_affiliations',
+            'facility_product_types',
+            'facility_production_types',
+            'facility_parent_company',
+        ]
+
+        return {
+            'updated_at': str(claim.updated_at),
+            'action': FacilityHistoryActions.CLAIM_UPDATE,
+            'detail': 'Facility {} claim public data was updated'.format(
+                claim.facility.id,
+            ),
+            'changes': {
```

Comment:
If a non-public field is updated, the history record looks like this:

<img width="614" alt="Screen Shot 2019-10-03 at 10 37 29 AM" src="https://user-images.githubusercontent.com/17363/66150458-14679d00-e5ca-11e9-8d7e-c9d0b42a9312.png">

This looks a bit broken. Consider suppressing the entire history entry if the changes dict is empty.